source/material_model/viscoelastic.cc

```diff
@@ -0,0 +1,700 @@
+                             "When more than one compositional field is present at a point "
+                             "with different viscosities, we need to come up with an average "
+                             "viscosity at that point.  Select a weighted harmonic, arithmetic, "
+                             "geometric, or maximum composition.");
+          prm.declare_entry ("Elastic shear moduli", "75.0e9",
+                             Patterns::List(Patterns::Double(0)),
+                             "List of elastic shear moduli, $G$, "
+                             "for background material and compositional fields, "
+                             "for a total of N+1 values, where N is the number of compositional fields. "
+                             "The default value of 75 GPa is representative of mantle rocks. Units: Pa.");
+          prm.declare_entry ("Use fixed elastic time step", "false",
+                             Patterns::Bool (),
+                             "Select whether the material time scale in the viscoelastic constitutive"
+                             "relationship uses the regular numerical time step or a separate fixed"
+                             "elastic time step throughout the model run. The fixed elastic time step"
+                             "is always used during the initial time step. Note that there is no"
+                             "physical motivation for using an elastic time step that differs from"
+                             "the numerical time step. As such, this option should only be used in"
+                             "reproducing previous studies or if the numerical time step is a"
+                             "constant value, which can be enforced through the CFL condition and"
+                             "maximum time step parameter. When using a fixed elastic time step that"
+                             "differs the numerical time step it is strongly recommended that the stress"
+                             "averaging scheme is also applied");
+          prm.declare_entry ("Fixed elastic time step", "1.e3",
+                             Patterns::Double (0),
+                             "The fixed elastic time step $dte$. Units: $yr$.");
+          prm.declare_entry ("Use stress averaging","false",
+                             Patterns::Bool (),
+                             "Whether to apply a stress averaging scheme to account for differences"
+                             "between the fixed elastic time step and numerical time step.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    Viscoelastic<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      //not pretty, but we need to get the number of compositional fields before
+      //simulatoraccess has been initialized here...
```

Comment:
This comment is outdated, and you can remove the subsection below. `this->n_compositional_fields()` should work just fine anywhere you need it.